File: 323-getTimeofMovingKeys_HashTable.py
#given a set of chars in our keyboard and one string which want to get the length
#used dictionary to record each chars index
def getTime(keys,s):
    d={}
    for i,c in enumerate(keys):
        d[c]=i
    ans=d[s[0]]
    
    for i in range(1,len(s)):
        ans+=abs(d[s[i]]-d[s[i-1]])
        
    return ans

keyboard = "abcdefghijklmnopqrstuvwxyz"
word = "cba"

print(getTime(keyboard,word))

File: 323-getTimeofMovingKeys_HashTable.cpp
//given a keyboards and one string,to return the total count of keys

#include<string>
#include<vector>
#include<iostream>
#include<map>

using STR=std::string;
using MAP=std::map<char,int>;


int getCount(const STR &keys, const STR &word)
{
    int kLen=keys.size(),wLen=word.size();
    if(!kLen ||!wLen)
        return 0;

    MAP m{};
    for(int i=0;i<kLen;++i)
    {
        m[keys[i]]=i;
    }

    int ans=m[word[0]];
    for(int i=1;i<wLen;++i)
        ans+=std::abs(m[word[i]]-m[word[i-1]]);
    return ans;
}

int main()
{
    STR keyboard = "abcdefghijklmnopqrstuvwxyz";
    STR word = "cba";
    std::cout<<getCount(keyboard,word);
    return 0;
}
